src/helper/remote_config/client.cpp

```diff
@@ -28,25 +33,48 @@ config_path config_path::from_path(const std::string &path)
 }
 
 client::client(std::unique_ptr<http_api> &&arg_api, service_identifier sid,
-    remote_config::settings settings, const std::vector<product> &products,
-    std::vector<protocol::capabilities_e> capabilities)
+    remote_config::settings settings, const std::vector<product> &products)
     : api_(std::move(arg_api)), id_(dds::generate_random_uuid()),
-      sid_(std::move(sid)), settings_(std::move(settings)),
-      capabilities_(std::move(capabilities))
+      sid_(std::move(sid)), settings_(std::move(settings))
 {
     for (auto const &product : products) {
         products_.insert(std::pair<std::string, remote_config::product>(
             product.get_name(), product));
+        capabilities_ |= product.get_capabilities();
     }
 }
 
 client::ptr client::from_settings(const service_identifier &sid,
     const remote_config::settings &settings,
-    std::vector<remote_config::product> &&products,
-    std::vector<protocol::capabilities_e> capabilities)
+    std::optional<bool> client_enable_configuration,
```

Comment:
It might be best o not propagate this but rather something more obvious:
- `dynamic_enablement` = true | false
- `dynamic_engine` = true | false